packages/starlight/integrations/code-rtl-support.ts

```diff
@@ -0,0 +1,31 @@
+import type { Root } from 'hastscript/lib/core';
+import { CONTINUE, SKIP, visit } from 'unist-util-visit';
+
+/**
+ * rehype plugin that adds `dir` attributes to `<code>` and `<pre>`
+ * elements that don’t already have them.
+ *
+ * `<code>` will become `<code dir="auto">`
+ * `<pre>` will become `<pre dir="ltr">`
+ *
+ * `<code>` _inside_ `<pre>` is skipped, so respects the `ltr` on its parent.
+ *
+ * Reasoning:
+ * - `<pre>` is usually a code block and code should be LTR even in an RTL document
+ * - `<code>` is often LTR, but could also be RTL. `dir="auto"` ensures the bidirectional
+ *   algorithm treats the contents of `<code>` in isolation and gives its best guess.
+ */
+export function rehypeRtlCodeSupport() {
+	return () => (root: Root) => {
+		visit(root, 'element', (el) => {
+			if (el.tagName === 'pre' || el.tagName === 'code') {
+				el.properties = el.properties || {};
```

Comment:
I always wanted to suggest this new syntax
```suggestion
				el.properties ||= {};
```

Reply:
👀

Do you happen to know the Node version that was added in? We safe to use it in 16+?

Reply:
`??=` & `||=` are both 15.0.

Reply:
i'd go with `??=` which is more usual to see nullish coalescing, but i guess that it is also a matter of choice.